src/medley/core.cljc

```diff
@@ -357,6 +357,64 @@
   ([pred coll]
    (rest (drop-while (complement pred) coll))))
 
+(defn partition-after
+  "Returns a lazy sequence of partitions, splitting after `(pred item)` returns
+  true. Returns a transducer when no collection is provided."
+  ([pred]
+   (fn [rf]
+     (let [part (volatile! [])
```

Comment:
I notice that in clojure.core an ArrayList is used. Is there a reason for avoiding that here? Is it just ClojureScript compatibility?

Reply:
Yes, exactly. I guess we could use reader conditionals to circumvent? Main downside is readability I suppose.

Reply:
I think conditionals are the way to go here. It does make it less readable, but one of Medley's goals is to be as fast or faster than Clojure core where possible.

Reply:
That's now implemented with reader conditionals. lmk what you think.